public/app/plugins/datasource/prometheus/datasource.tsx

```diff
@@ -455,6 +455,7 @@ export class PrometheusDatasource
       let fullOrPartialRequest: DataQueryRequest<PromQuery>;
       let requestInfo: CacheRequestInfo | undefined = undefined;
       if (this.hasIncrementalQuery) {
+        request.appendRequestId = true;
```

Comment:
To cover expressions, I think this part should be copied to `public/app/features/expressions/ExpressionDatasource.ts` as well. Or maybe even having a thing for all datasources. But I see you want to have it only for Prometheus. Is that right?

Reply:
lets not copy it anyway... but move it so it is not prometheus specific.